lib/instrument/transformer/dtx-encode.js

```diff
@@ -0,0 +1,32 @@
+import Stream from 'stream';
+import {DTXMessage} from '../headers';
+
+
+class DTXEncoder extends Stream.Transform {
+
+  constructor () {
+    super({ objectMode: true });
+  }
+
+  _transform (data, encoding, onData) {
+    this.push(this._encode(data), 'buffer');
+    onData();
+  }
+
+  _encode (data) {
+    const {sync, channelID, selector, auxiliaries, identifier} = data;
+    const dtx = new DTXMessage();
+    dtx._identifier = identifier;
+    dtx._channelCode = channelID;
+    dtx._selector = selector;
+    dtx._expectsReply = sync;
+    if (auxiliaries.constructor.name === 'DTXMessageAuxBuffer') {
```

Comment:
it would be more appropriate to use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/instanceof